kafka/config.go

```diff
@@ -0,0 +1,192 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"log/syslog"
+	"strings"
+	"time"
+
+	kafkalib "github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Supported auth types
+const (
+	AuthTypePlain    = "plain"
+	AuthTypeSCRAM256 = "scram-sha256"
+	AuthTypeSCRAM512 = "scram-sha512"
+)
+
+// DefaultLogLevel is the log level Kafka producers/consumers will use if non set.
+const DefaultLogLevel = logrus.ErrorLevel
+
+// Config holds all the configuration for this package.
+type Config struct {
+	Brokers   []string       `json:"brokers"`
+	Topic     string         `json:"topic"`
+	Producer  ProducerConfig `json:"producer" envconfig:"producer"`
+	Consumer  ConsumerConfig `json:"consumer" envconfig:"consumer"`
+	AuthType  string         `json:"auth" split_words:"true"`
+	User      string         `json:"user"`
+	Password  string         `json:"password"`
+	CAPEMFile string         `json:"ca_pem_file"`
+	LogLevel  string         `json:"log_level" split_words:"true"`
+}
+
+// baseKafkaConfig provides the base config that applies to both consumers and producers.
+func (c Config) baseKafkaConfig() *kafkalib.ConfigMap {
+	logrusToSylogLevelMapping := map[logrus.Level]syslog.Priority{
```

Comment:
this could be pulled out into a var outside the method scope?

Reply:
What would be the benefit?

Reply:
you wouldn't re-instantiate it at every invocation, minor b/c this is only invoked once-ish.

Reply:
Feel free to ignore